components/butterfly/src/error.rs

```diff
@@ -33,6 +33,9 @@ pub enum Error {
     BadMessage(String),
     CannotBind(io::Error),
     DatFileIO(PathBuf, io::Error),
+    GossipPullSocketReceiveError(String),
```

Comment:
Hadn't we agreed on using the failure crate? How does this play into that?

/cc @zeenix

Reply:
I would need to analyze first, how the errors are used in butterfly and convert the whole component to it first. There are several ways of using failure, so I stated I won't waste time on it.